examples/basic/gpu_shared_memory.c

```diff
@@ -0,0 +1,245 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <ze_api.h>
+
+#include "umf/memory_pool.h"
+#include "umf/pools/pool_disjoint.h"
+#include "umf/providers/provider_level_zero.h"
+
+// This function initializes the Level Zero driver and creates a context in the
+// first GPU device found.
+int level_zero_init(ze_driver_handle_t *driver, ze_device_handle_t *device,
```

Comment:
In PR #418 I created some utility functions on top of the L0 API. We need to coordinate and decide how to re-use them among examples and tests where L0 provider is involved.